### PR TITLE
Added improved help texts for config command

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

**Comment by petersutter** [comment]
suggestion for next time: 

split into two commits:
- help text improvements
- the generated stuff

File: pkg/cmd/config/delete_garden.go [hunk 1]
```diff
@@ -26,9 +26,10 @@ func NewCmdConfigDeleteGarden(f util.Factory, ioStreams util.IOStreams) *cobra.C
 		},
 	}
 	cmd := &cobra.Command{
-		Use:               "delete-garden",
-		Short:             "delete Garden from gardenctl configuration",
-		Long:              "Delete Garden from gardenctl configuration. E.g. \"gardenctl config delete-garden my-garden\" to delete my-garden",
+		Use:   "delete-garden",
+		Short: "delete the specified Garden from the gardenctl configuration",
+		Example: `#delete my-garden
```
Comment (petersutter) [inline pkg/cmd/config/delete_garden.go:31]:
```suggestion
		Example: `# delete my-garden
```

File: pkg/cmd/config/set_garden.go [hunk 2]
```diff
@@ -29,9 +29,21 @@ func NewCmdConfigSetGarden(f util.Factory, ioStreams util.IOStreams) *cobra.Comm
 		},
 	}
 	cmd := &cobra.Command{
-		Use:               "set-garden",
-		Short:             "modify or add Garden to gardenctl configuration",
-		Long:              "Modify or add Garden to gardenctl configuration. E.g. \"gardenctl config set-garden my-garden --kubeconfig ~/.kube/kubeconfig.yaml\" to configure or add a garden with identity 'my-garden'",
+		Use:   "set-garden",
+		Short: "modify or add a Garden to the gardenctl configuration",
+		Long: `Modify or add a Garden to the gardenctl configuration.
+A valid Garden configuration consists of a name (required), kubeconfig path (required), an optional context as well as any number of patterns.
```
Comment (petersutter) [inline pkg/cmd/config/set_garden.go:35]:
```suggestion
A valid Garden configuration consists of a name (required), kubeconfig path (required), a context as well as any number of patterns.
```
I think we do not necessarily need to mention that it's optional (as it's not required)

**Comment by grolu** [comment]
>suggestion for next time:
>split into two commits

Yes, I thought about it. Maybe the best approach is even to not generate until we applied all PR Feedback.

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**